netket/models/deepset.py

```diff
@@ -0,0 +1,129 @@
+from typing import Union, Tuple, Any
+
+import jax
+from jax import numpy as jnp
+from flax import linen as nn
+from netket.utils.types import NNInitFunc
+from jax.nn.initializers import (
+    zeros,
+    ones,
+    lecun_normal,
+)
+
+
+class DeepSet(nn.Module):
+    r"""Implements an equivariant version of the DeepSets architecture
+    given by (https://arxiv.org/abs/1703.06114)
+
+    .. math ::
+
+        f(x_1,...,x_N) = \rho\left(\sum_i \phi(x_i)\right)
+
+    that is suitable for the simulation of periodic systems.
+    Additionally one can add a cusp condition by specifying the
+    asymptotic exponent.
+    For helium the Ansatz reads:
+
+    .. math ::
+
+        \psi(x_1,...,x_N) = \rho\left(\sum_i \phi(d_{\sin}(x_i,x_j))\right) \cdot \exp\left[-\frac{1}{2}\left(b/d_{\sin}(x_i,x_j)\right)^5\right]
+
+    """
+    L: jnp.float64
+    """boxsize"""
+    sdim: int
+    """number of spatial dimensions"""
+
+    layers_phi: int
+    """Number of layers in phi network"""
+    layers_rho: int
+    """Number of layers in rho network"""
+
+    features_phi: Union[Tuple, int]
+    """Number of features in each layer for phi network."""
+    features_rho: Union[Tuple, int]
+    """Number of features in each layer for rho network."""
+
+    cusp_exponent: int = 0.0
```

Comment:
Can we make this an `Optional[int]` and have it None by default instead of 0? 
And in the apply method check that if cusp is None, then don't apply it (so it won't compute distances, etc)